### PR TITLE
Input validation for single simulator form

### PR DESCRIPTION
**Pull request by hasiniravindi**
**Purpose**
Users should not be able to type invalid characters at all.
Fixes :#397

**Security checks**
Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
Ran FindSecurityBugs plugin and verified report? yes
Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

**Test environment**
Tested on Oracle JDK 1.8

### CODE REVIEW

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/simulator.js [hunk 1]
```diff
@@ -768,10 +768,38 @@ define(['jquery', 'log', './simulator-rest-client', 'lodash', './open-siddhi-app
         $attributes.each(
             function () {
                 self.addRuleForAttribute(this);
+                var dataType = $(this).attr("data-type");
+                $(this).keypress(function(e) {
+                    var userInput = e.key;
+                    if (userInput != "Delete" && userInput != "Backspace" &&
```
[the remark this answers is not in the document]
Comment (dnwick) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/simulator.js]: Also remove specific checks

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/simulator.js [hunk 2]
```diff
@@ -768,10 +768,38 @@ define(['jquery', 'log', './simulator-rest-client', 'lodash', './open-siddhi-app
...
+                    }
+                });
             }
         );
     };
 
+//add real time validation for an attribute based on the attribute type
+    self.realTimeValidate = function(dataType, userInput) {
+        var validDataType = false;
+        var intLongRegexMatch = /^-?[\d]*$/;
```
Comment (dnwick) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/simulator.js]: use constants

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/simulator.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hasiniravindi) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/simulator.js]: fixed in #ef4ecd9

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/simulator.js (hunk 2 above)
Comment (hasiniravindi) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/simulator.js]: fixed in #ef4ecd9

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/constants.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+define(function() {
+    "use strict"; // JS strict mode
+
+    /**
+     * Constants used by the tool - editor
+     */
+    var constants = {
+        ALLOWED_KEYS: ["Delete", "ArrowRight", "ArrowLeft", "Backspace"],
```
[the remark this answers is not in the document]
Comment (hasiniravindi) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/constants.js]: fixed in #dde2d07

**Pull request merged**